### ISSUE TITLE
User acct deletion: modal overlay to confirm/cancel user deletion. 

### ISSUE THREAD

**Issue by tobinmori** [mdn/kuma#6071]
**Parent ticket:**
https://github.com/mdn/sprints/issues/1922

**Description:**
Overlay appears on MDN profile page when a logged-in MDN user clicks on 'Delete Account' link. 

**Acceptance Criteria:**
- [x] Modal needs to work on all devices. (for mobile - will go to a separate screen)
- [x] Clicking on 'Cancel' closes modal
- [x] Clicking on 'Yes, Delete my Account' executes deletion and closes modal. 

**Design:**
https://mozilla.invisionapp.com/share/Y8SGMKQBCW3#/screens/370705983_Account_-_Profile_-_Delete_Account_-no_Contrib-

**Comment by tobinmori** [comment]
@atopal where should the 'Delete Account' link (that launches the modal) appear on the profile page? (I don't see it on the design)

**Comment by schalkneethling** [comment]
> @atopal where should the 'Delete Account' link (that launches the modal) appear on the profile page? (I don't see it on the design)

That will be at the bottom of the edit profile page. I will spin up a separate issue for the UI portion of this work.

**Comment by schalkneethling** [comment]
Never mind, this is that issue ;)

**Comment by tobinmori** [comment]
Hi @schalkneethling, a couple questions for you:

- re: status. is this currently still waiting to be done?
- is this still blocked by the sign-in landing page issue?

**Comment by schalkneethling** [comment]
> Hi @schalkneethling, a couple questions for you:
> 
> * re: status. is this currently still waiting to be done?
> * is this still blocked by the sign-in landing page issue?

This is ready to move to Ready for Dev. One question I have is, while I am in there in the weeds of the edit profile view, why not go ahead and start cleaning up?

There are a *lot* of fields we know we will not be using going forward so, the sooner we start cleaning up the better i.e. follow on work will be much quicker.

I can also then take care of removing jQueryUI, added the React-based header, and start updating the UI to use the new styling for button etc.

Let me know @tobinmori - Thanks!

**Comment by tobinmori** [comment]
I'm all for cleanup @schalkneethling! 

Let me defer this to @atopal.  Are you ok with this? (as it might delay this task a bit, but mean faster movement later. . .)

**Comment by tobinmori** [comment]
The other thing that we discussed @schalkneethling yesterday is that the existing design for the user deletion needs to be revamped. Let's talk about that when you have a chance.

**Comment by tobinmori** [comment]
@atopal to test out existing functionality and verify whether to keep it, or update it.

**Comment by atopal** [comment]
I looked into this.

1. I don't know what the flow looks like for someone without any contributions. Does it immediately delete the account without any further message after the first "delete account" link click?
2. I don't know what happens when you actually click "delete my account" on the follow up page.

Can someone provide screenshots?

How this is different from the original design:
1. Message about deleting your account.
2. Scary red button for deleting account
3. Keep attribution doesn't show the username that would be attributed
4. Lists the number of contributions
5. Missing the explanation for why we are asking about attribution.

We can talk about adjusting what we have, but I think it's easier to just go ahead and implement the original design.

**Comment by peterbe** [comment]
Quick "answer" is that it looks like the screenshots in https://github.com/mdn/kuma/pull/6149
And from skimming https://github.com/mdn/kuma/blob/master/kuma/users/jinja2/users/user_delete.html it appears that if you have 0 contributions it'll say 
```html
     <p>We're sorry to see you go.</p>

        <p>
          You have made 0 wiki revisions.
        </p>

<button type="submit" class="button positive">Delete my account</button>
```

I'm guessing this is for Schalk. But it'll be obvious once he sets aside the time to work on this further.

**Comment by schalkneethling** [comment]
Note to self: https://github.com/mdn/sprints/issues/1635

**Comment by schalkneethling** [comment]
Links to some of the designs from Nick:

* [Edit profile page with `Delete Account` button](https://mozilla.invisionapp.com/share/Y8SGMKQBCW3#/screens/372750124_Account_-_Profile_-_Filled)
* [No contributions made modal](https://mozilla.invisionapp.com/share/Y8SGMKQBCW3#/screens/370705983_Account_-_Profile_-_Delete_Account_-no_Contrib-)
* [1+ Contributions made step 1](https://mozilla.invisionapp.com/share/Y8SGMKQBCW3#/screens/370705982_Account_-_Profile_-_Delete_Account_-has_Contrib-)
* [1+ Contributions made step 2](https://mozilla.invisionapp.com/share/Y8SGMKQBCW3#/screens/370705981_Account_-_Profile_-_Delete_Account_-_Filled_-has_Contrib-)
